src/coreclr/vm/mlinfo.cpp

```diff
@@ -2544,7 +2544,13 @@ void MarshalInfo::SetupArgumentSizes()
     const unsigned targetPointerSize = TARGET_POINTER_SIZE;
     const bool pointerIsValueType = false;
     const bool pointerIsFloatHfa = false;
+#ifdef TARGET_WASM
+    // Wasm uses the interpreter which uses a pointer agnostic stack slot size.
+    // Therefore, we only need assert it is sufficiently large.
+    _ASSERTE(targetPointerSize <= StackElemSize(TARGET_POINTER_SIZE, pointerIsValueType, pointerIsFloatHfa));
+#else
     _ASSERTE(targetPointerSize == StackElemSize(TARGET_POINTER_SIZE, pointerIsValueType, pointerIsFloatHfa));
```

Comment:
I am not sure why this assert is useful. How is it relevant to the code that follows?

I think it would be better to delete it instead of ifdefing it out for wasm.